models/Tag.js

```diff
@@ -75,17 +76,22 @@ module.exports = function(sequelize, DataTypes) {
   Tag.deleteModelInstance = function(id, user) {
     util.deleteModelInstance(id, user);
   };
-  
+
   Tag.deleteFromId = async function(id, user) {
     var tag = await this.findOne({where: {id: id}});
     if (tag == null) {
       return false;
     }
-    if (tag.taggerId === user.id) {
-      await tag.destroy();
-      return true;
+    const recording = await Recording.get(
+      user,
+      tag.RecordingId,
+      Recording.Perms.TAG,
+    );
+
+    if(recording == null){
+      return false;
     }
-    else {return false;}
+    return tag.destroy();
```

Comment:
shouldn't this be:
```js
await tag.destroy();
return true;
```
?

Otherwise sometimes it returns false and sometimes it returns a promise which returns who knows what.

Reply:
The promise should return 1 if successful and 0 if unsuccessful, I figured because the function is async a promise is an expected result. Seems strange to have all these async functions with awaits in them?

Reply:
It's totally fine for an async function to return a value. You can only use `await` in an async function. When an async function is called (often with await) it means the JS runtime is allowed to run other things during the function call (e.g. when it calls await on other functions).

An async function can also return promises but then the return value ends up being the return value of that promise. In this case `tag.destroy()` returns a promise which returns the number of deleted rows. It's messy to sometimes return `false`, sometimes `1` and sometimes `0`.

Also, it should be considered an error if a tag was to be deleted that was already deleted. This is success as the intent of the operation has happened.

Reply:
Turns out I was reading the wrong doc for destroy anyway. It just returns a promise.